changes to build 5.3 with gfortran 11

jdhughes-dev opened the pull request.

jdhughes-dev committed:
changes to build 5.3 with gfortran 11

jdhughes-dev committed:
Add allocated(Irr_type) check before setting

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
ci: add gfortran and intel build GitHub Actions workflows

jdhughes-dev committed:
Modify encoding of snowcomp.f90 and soltab.f90 to ansi to fix OneAPI
build issue on Windows.

jdhughes-dev committed:
Update windows-2019 to windows-2022 VM

jdhughes-dev committed:
OneAPI for Windows does not support Visual Studio 2022 yet

jdhughes-dev committed:
Update meson.build with version number

jdhughes-dev commented:
@rsregan This is ready to go in my opinion. I would squash and merge to get rid of all of the commits to get GitHub actions up and running.

Changes include:
- Removing intrinsic SNGL() conversion for variables that are single precision (gfortran issue)
- Adding continuation & for lines in obs.f90 and soilzone.f90 exceeding 132 character (gfortran issue)
- Add allocated(Irr_type) check before setting to zero (gfortran issue)
- Modified snowcomp.f90 and soltab.f90 encoding to ansi to fix OneAPI build issue on Windows
- Add meson build files
- Add gfortran and ifort workflows to build PRMS on Linux, macOS, and Windows

You will need to enable actions on GitHub before merging this PR. You can look at my fork
to see the latest [gfortran](https://github.com/jdhughes-usgs/prms/actions/runs/1785678875) and [ifort](https://github.com/jdhughes-usgs/prms/actions/runs/1785678876) GitHub Actions workflow runs.

rsregan commented:
I've been unable to figure out what was going with snowcomp and soltab for years, as they would sometimes get in an infinite loop when compiled with gfortran. They would sometimes compile and sometimes not.

I didn't get an issue with the SNGL conversion with gfortran.

how did you figure out these issues? What compile options are you using?


R. Steve Regan
U.S. Geological Survey
303-236-5008 (office)
PO Box 25046, MS 413 Bldg 53
Denver Federal Center
Lakewood, CO 80225-0046
success is rarely about finding the right solution to a problem — it’s more about finding the right problem to solve

________________________________
From: jdhughes-usgs ***@***.***>
Sent: Wednesday, February 2, 2022 12:35 PM
To: nhm-usgs/prms ***@***.***>
Cc: Regan, Robert S ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [nhm-usgs/prms] changes to build 5.3 with gfortran 11 (PR #21)




 This email has been received from outside of DOI - Use caution before clicking on links, opening attachments, or responding.



@rsregan<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frsregan&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Z4SFskLxra2c2CP7Mn8omZTPHE3dXp4u3pRh%2Ftmtn6Y%3D&reserved=0> This is ready to go in my opinion. I would squash and merge to get rid of all of the commits to get GitHub actions up and running.

Changes include:

  *   Removing intrinsic SNGL() conversion for variables that are single precision (gfortran issue)
  *   Adding continuation & for lines in obs.f90 and soilzone.f90 exceeding 132 character (gfortran issue)
  *   Add allocated(Irr_type) check before setting to zero (gfortran issue)
  *   Modified snowcomp.f90 and soltab.f90 encoding to ansi to fix OneAPI build issue on Windows
  *   Add meson build files
  *   Add gfortran and ifort workflows to build PRMS on Linux, macOS, and Windows

You will need to enable actions on GitHub before merging this PR. You can look at my fork
to see the latest gfortran<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjdhughes-usgs%2Fprms%2Factions%2Fruns%2F1785678875&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=FAYxzy%2Bh3%2FoWYgBrh8u%2BgG5HUYSjl%2FjfIWVT5X819a8%3D&reserved=0> and ifort<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjdhughes-usgs%2Fprms%2Factions%2Fruns%2F1785678876&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=GWDCXSyJq0yRwv98dm31RZAqqYRQGSKD%2BjMpZz28QmA%3D&reserved=0> GitHub Actions workflow runs.

—
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnhm-usgs%2Fprms%2Fpull%2F21%23issuecomment-1028288583&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=dV43HbKjTiWKgE%2FVrcMY2ZfgZTvZzsTk6crVrleaZpY%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FACLN2FNHAG3UF67G34E5FPDUZGBPJANCNFSM5NHJHLPA&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=uXkWjlJn5P%2F7fvVh8fzNwat2ifz7d0lijcXT9FvNFig%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=paL%2BNuC1PTO4svpTib%2BJV7tx5RhAf3TNaCn8%2FZMLShs%3D&reserved=0> or Android<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Bi4mO6f7qlxePrlQkErf44yB4A4pGkZh%2FrHRHPIXOvc%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

jdhughes-dev commented:
For snowcomp.f90 and soltab.f90 I got this error message

../prms\soltab.f90(1): error #5078: Unrecognized token '?' skipped
70<https://github.com/jdhughes-usgs/prms/runs/5040431256?check_suite_focus=true#step:12:70>﻿!***********************************************************************
71<https://github.com/jdhughes-usgs/prms/runs/5040431256?check_suite_focus=true#step:12:71>^



So I figured there was some hidden trash in the file. I used notepad++ so see that the encoding was not ascii, changed it ascii, and the problem was fixed.

I got an error about message for SNGL on macOS with gfortran-11. Maybe it is an error with 11 but not earlier versions. It could also be the compiler options. You can see the compiler options I used in the meson.build file in the root directory.



From: Steve Regan ***@***.***>
Date: Thursday, February 3, 2022 at 1:16 PM
To: nhm-usgs/prms ***@***.***>
Cc: Hughes, Joseph D ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [nhm-usgs/prms] changes to build 5.3 with gfortran 11 (PR #21)



 This email has been received from outside of DOI - Use caution before clicking on links, opening attachments, or responding.


I've been unable to figure out what was going with snowcomp and soltab for years, as they would sometimes get in an infinite loop when compiled with gfortran. They would sometimes compile and sometimes not.

I didn't get an issue with the SNGL conversion with gfortran.

how did you figure out these issues? What compile options are you using?


R. Steve Regan
U.S. Geological Survey
303-236-5008 (office)
PO Box 25046, MS 413 Bldg 53
Denver Federal Center
Lakewood, CO 80225-0046
success is rarely about finding the right solution to a problem — it’s more about finding the right problem to solve

________________________________
From: jdhughes-usgs ***@***.***>
Sent: Wednesday, February 2, 2022 12:35 PM
To: nhm-usgs/prms ***@***.***>
Cc: Regan, Robert S ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [nhm-usgs/prms] changes to build 5.3 with gfortran 11 (PR #21)




This email has been received from outside of DOI - Use caution before clicking on links, opening attachments, or responding.



@rsregan<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frsregan&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Z4SFskLxra2c2CP7Mn8omZTPHE3dXp4u3pRh%2Ftmtn6Y%3D&reserved=0> This is ready to go in my opinion. I would squash and merge to get rid of all of the commits to get GitHub actions up and running.

Changes include:

* Removing intrinsic SNGL() conversion for variables that are single precision (gfortran issue)
* Adding continuation & for lines in obs.f90 and soilzone.f90 exceeding 132 character (gfortran issue)
* Add allocated(Irr_type) check before setting to zero (gfortran issue)
* Modified snowcomp.f90 and soltab.f90 encoding to ansi to fix OneAPI build issue on Windows
* Add meson build files
* Add gfortran and ifort workflows to build PRMS on Linux, macOS, and Windows

You will need to enable actions on GitHub before merging this PR. You can look at my fork
to see the latest gfortran<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjdhughes-usgs%2Fprms%2Factions%2Fruns%2F1785678875&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=FAYxzy%2Bh3%2FoWYgBrh8u%2BgG5HUYSjl%2FjfIWVT5X819a8%3D&reserved=0> and ifort<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjdhughes-usgs%2Fprms%2Factions%2Fruns%2F1785678876&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=GWDCXSyJq0yRwv98dm31RZAqqYRQGSKD%2BjMpZz28QmA%3D&reserved=0> GitHub Actions workflow runs.

—
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnhm-usgs%2Fprms%2Fpull%2F21%23issuecomment-1028288583&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=dV43HbKjTiWKgE%2FVrcMY2ZfgZTvZzsTk6crVrleaZpY%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FACLN2FNHAG3UF67G34E5FPDUZGBPJANCNFSM5NHJHLPA&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=uXkWjlJn5P%2F7fvVh8fzNwat2ifz7d0lijcXT9FvNFig%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=paL%2BNuC1PTO4svpTib%2BJV7tx5RhAf3TNaCn8%2FZMLShs%3D&reserved=0> or Android<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Crsregan%40usgs.gov%7C90788a5eba964433996408d9e6832f18%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637794273370910049%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Bi4mO6f7qlxePrlQkErf44yB4A4pGkZh%2FrHRHPIXOvc%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>


—
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnhm-usgs%2Fprms%2Fpull%2F21%23issuecomment-1029316008&data=04%7C01%7Cjdhughes%40usgs.gov%7C445bd9e78d0d40c85ac208d9e749b645%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637795126037961691%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=p4UpMZsfVaRn7x%2BAzNK5hwowqgKQuaXsMV0Fra6pSKs%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAP2HGTBHEXVT7TWEQ7MN7LUZLIAFANCNFSM5NHJHLPA&data=04%7C01%7Cjdhughes%40usgs.gov%7C445bd9e78d0d40c85ac208d9e749b645%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637795126037961691%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=%2BZte9N9KEvYcx44YTW7aD1nuM0YGpzSYaZ9wVumsxF8%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cjdhughes%40usgs.gov%7C445bd9e78d0d40c85ac208d9e749b645%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637795126037961691%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=j1XIcJg6UvfGCxyTTLz12lruhJTQi5ozVO9yltcou34%3D&reserved=0> or Android<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cjdhughes%40usgs.gov%7C445bd9e78d0d40c85ac208d9e749b645%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637795126037961691%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=OBfq1rJKxiXI4h1OMhRh%2BqHcPhAUbAzKMuO6ZOqwtHI%3D&reserved=0>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
